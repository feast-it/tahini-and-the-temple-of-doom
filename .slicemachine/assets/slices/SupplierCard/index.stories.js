import MyComponent from '../../../../slices/SupplierCard';

export default {
  title: 'slices/SupplierCard'
}


export const _Default = () => <MyComponent slice={{"variation":"default","version":"sktwi1xtmkfgx8626","items":[{}],"primary":{"title":[{"type":"heading1","text":"Positive","spans":[]}],"tagline":[{"type":"paragraph","text":"Duis consectetur exercitation ipsum reprehenderit. Minim voluptate cillum quis dolore aliqua mollit ad minim aliquip.","spans":[]}],"image":{"link_type":"Media","name":"mock","kind":"image","url":"https://source.unsplash.com/random","size":"10kB"},"rating":2},"id":"_Default","slice_type":"supplier_card"}} />
_Default.storyName = ''
