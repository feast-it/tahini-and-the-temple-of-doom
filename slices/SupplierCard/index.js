import React from "react";
import { PrismicRichText, PrismicImage } from "@prismicio/react";
import styled from "styled-components";

const Card = styled.div`
  border: 2px solid gray;
  border-radius: 4px;
`;
const SectionContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;

const TextContainer = styled.div``;

/**
 * @typedef {import("@prismicio/client").Content.SupplierCardSlice} SupplierCardSlice
 * @typedef {import("@prismicio/react").SliceComponentProps<SupplierCardSlice>} SupplierCardProps
 * @param { SupplierCardProps }
 */
const SupplierCard = ({ slice }) => (
  <Card>
    <PrismicImage field={slice.primary.image} />
    <SectionContainer>
      <TextContainer>
        <PrismicRichText field={slice.primary.title} />
        <PrismicRichText field={slice.primary.tagline} />
      </TextContainer>
      <span>{slice.primary.rating}</span>
    </SectionContainer>
  </Card>
);

export default SupplierCard;
