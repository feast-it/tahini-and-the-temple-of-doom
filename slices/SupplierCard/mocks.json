[
  {
    "__TYPE__": "SharedSliceContent",
    "variation": "default",
    "primary": {
      "title": {
        "__TYPE__": "StructuredTextContent",
        "value": [
          {
            "type": "heading1",
            "content": {
              "text": "Positive"
            }
          }
        ]
      },
      "tagline": {
        "__TYPE__": "StructuredTextContent",
        "value": [
          {
            "type": "paragraph",
            "content": {
              "text": "Duis consectetur exercitation ipsum reprehenderit. Minim voluptate cillum quis dolore aliqua mollit ad minim aliquip."
            }
          }
        ]
      },
      "image": {
        "__TYPE__": "LinkContent",
        "value": {
          "id": "mock_Media_link",
          "kind": "image/jpeg",
          "name": "mock_name",
          "url": "https://source.unsplash.com/daily",
          "size": "0",
          "__TYPE__": "FileLink"
        }
      },
      "rating": {
        "__TYPE__": "FieldContent",
        "value": "2",
        "type": "Number"
      }
    },
    "items": [
      {
        "__TYPE__": "GroupItemContent",
        "value": []
      }
    ]
  }
]