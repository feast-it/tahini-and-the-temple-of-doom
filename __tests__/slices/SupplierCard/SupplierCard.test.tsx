import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import SupplierCard, { SupplierCardProps } from "../../../slices/SupplierCard";

describe("Home", () => {
  it("renders a heading", () => {
    const props: SupplierCardProps = {
      slice: {
        items: [],
        primary: {
          image: { link_type: "linkType" },
          rating: 2,
          tagline: [{ type: "heading1", text: "Lorem Ipsum", spans: [] }],
          title: [],
        },
        slice_type: "supplier_card",
        variation: "default",
        version: "",
        slice_label: null,
      },
      context: {},
      index: 1,
      slices: [],
    };

    const component = render(<SupplierCard {...props} />);

    expect(component.baseElement).toMatchSnapshot();
  });
});
