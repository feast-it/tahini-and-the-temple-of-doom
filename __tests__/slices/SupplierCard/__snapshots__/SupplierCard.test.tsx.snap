// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a heading 1`] = `
<body>
  <div>
    <div
      class="sc-beySbM kBUyjj"
    >
      <div
        class="sc-guDLRT dOErVi"
      >
        <div
          class="sc-dmyDGi"
        >
          <h1>
            Lorem Ipsum
          </h1>
        </div>
        <span>
          2
        </span>
      </div>
    </div>
  </div>
</body>
`;
